File: lib/JFFUI/UIViewController/UIViewController+PresentTopViewController.m
#import "UIViewController+PresentTopViewController.h"

@implementation UIViewController (PresentTopViewController)

- (void)presentTopViewController:(UIViewController *)viewControllerToPresent
                        animated:(BOOL)flag
                      completion:(void (^)(void))completion
{
    dispatch_async(dispatch_get_main_queue(), ^{
        UIViewController *presentingController = self.presentedViewController ?: self;
        [presentingController presentViewController:viewControllerToPresent
                                           animated:flag
                                         completion:completion];
    });
}

- (void)presentTopViewController:(UIViewController *)viewControllerToPresent
{
    [self presentTopViewController:viewControllerToPresent
                          animated:YES
                        completion:nil];
}

@end

File: iAsync.Swift/lib/JUI/JUI/UIViewController/UIViewController+PresentTopViewController.swift
//
//  UIViewController+PresentTopViewController.swift
//  JUI
//
//  Created by Vladimir Gorbenko on 19.03.15.
//  Copyright (c) 2015 EmbeddedSources. All rights reserved.
//

import Foundation

import JUtils

public extension UIViewController {

    func presentTopViewController(
        viewControllerToPresent: UIViewController,
        animated: Bool,
        completion: JSimpleBlock?
    )
    {
        let presentingController = presentedViewController ?? self
        presentingController.presentViewController(
            viewControllerToPresent,
            animated  : animated,
            completion: completion)
    }
    
    func presentTopViewController(viewControllerToPresent: UIViewController) {
        
        presentTopViewController(viewControllerToPresent, animated:true, completion:nil)
    }
}
